rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp

```diff
@@ -476,3 +522,58 @@ TEST_F(TestSubscriptionTopicStatisticsFixture, test_receive_stats_for_message_wi
     check_if_statistic_message_is_populated(msg);
   }
 }
+
+TEST_F(TestSubscriptionTopicStatisticsFixture, test_receive_stats_include_window_reset)
+{
+  // Create a MessageWithHeader publisher
+  auto msg_with_header_publisher = std::make_shared<TransitionMessageStampPublisher>(
+    kTestPubNodeName, kTestSubStatsTopic, kTransitionDuration, kInitialMessageAgeOffset);
+
+  // msg_with_header_subscriber has a topic statistics instance as part of its
+  // subscription this will listen to and generate statistics
+  auto msg_with_header_subscriber =
+    std::make_shared<MessageWithHeaderSubscriber>(kTestSubNodeName, kTestSubStatsTopic);
+  // Create a listener for topic statistics messages
+  auto statistics_listener = std::make_shared<rclcpp::topic_statistics::MetricsMessageSubscriber>(
+    "test_receive_stats_include_window_reset", "/statistics", kNumExpectedMessages);
+
+  rclcpp::executors::SingleThreadedExecutor ex;
+  ex.add_node(msg_with_header_publisher);
+  ex.add_node(statistics_listener);
+  ex.add_node(msg_with_header_subscriber);
+
+  // Spin and get future
+  ex.spin_until_future_complete(statistics_listener->GetFuture(), kTestDuration);
+
+  const auto received_messages = statistics_listener->GetReceivedMessages();
```

Comment:
Check that the number received is not zero or check against `kNumExpectedMessages`?

Reply:
Thanks.
I added this check.